jetty-webapp/src/main/java/org/eclipse/jetty/webapp/CachingWebAppClassLoader.java

```diff
@@ -116,12 +151,45 @@ public URL getResource(String name)
     public void clearCache()
     {
         _cache.clear();
+        _classCache.clear();
         _notFound.clear();
     }
-    
+
+    public void setUseCache(boolean useCache)
+    {
+            this.useCache = useCache;
+    }
+
     @Override
     public String toString()
     {
         return "Caching["+super.toString()+"]";
     }
+
+    /**
+     * Lifecycle listener, that can be used from {@link WebAppContext#addLifeCycleListener(LifeCycle.Listener)}
+     * to switch off the {@link #_cache} and {@link #_classCache} after web application has been started.
+     */
+    public static class ClearCacheLifeCycleListener extends AbstractLifeCycle.AbstractLifeCycleListener
+    {
+        private CachingWebAppClassLoader classLoader;
+
+        public ClearCacheLifeCycleListener(CachingWebAppClassLoader classLoader)
+        {
+            this.classLoader = classLoader;
+        }
+
+        @Override
+        public void lifeCycleStarting(LifeCycle event)
+        {
+            classLoader.setUseCache(true);
```

Comment:
You might want to interrogate the event to make sure that it belongs to the same `this.classLoader` before acting on it.

Reply:
Actually, scratch that, the `CachingWebAppClassLoader` isn't a `LifeCycle` so you can't interrogate it that way.

Reply:
> Actually, scratch that, the `CachingWebAppClassLoader` isn't a `LifeCycle` so you can't interrogate it that way.

Could you clarify this, please? I didn't catch what did you mean. As I understand I should remove this listener at all. Or maybe I should move it to sepate class?

Reply:
Depending on where you add this listener, it will get events for itself, and all children.

So you want to filter the events only to the ones you care about.

Example:

https://github.com/eclipse/jetty.project/blob/3af45f4c89adf000db6396f7b253d9b528cac301/jetty-cdi/cdi-websocket/src/main/java/org/eclipse/jetty/cdi/websocket/WebSocketCdiListener.java#L77-L98